src/SystemCommands-ClassCommands/SycCmAddSubclassCommand.class.st

```diff
@@ -34,7 +60,78 @@ SycCmAddSubclassCommand >> name [
 	^'New subclass'
 ]
 
+{ #category : 'accessing' }
+SycCmAddSubclassCommand >> newClassName [
+
+	^ newClassName 
+]
+
+{ #category : 'accessing' }
+SycCmAddSubclassCommand >> newClassName: aString [ 
+
+	newClassName := aString asSymbol
+]
+
 { #category : 'accessing' }
 SycCmAddSubclassCommand >> order [
 	^ 10100.1
 ]
+
+{ #category : 'accessing' }
+SycCmAddSubclassCommand >> packageName [
+
+	^ packageName
+]
+
+{ #category : 'accessing' }
+SycCmAddSubclassCommand >> packageName: anObject [
+
+	packageName := anObject
+]
+
+{ #category : 'accessing' }
+SycCmAddSubclassCommand >> packageNames [
+	"Answer a <Collection> of <Symbol> of the packages available to the new class"
+	
+	^ PackageOrganizer default packageNames
+]
+
+{ #category : 'preparation' }
+SycCmAddSubclassCommand >> prepareFullExecution [
+	
+	refactoringScopes := context refactoringScopes.
+	targetClass := context lastSelectedClass.
+	(StRefactoringAddClassPresenter on: self)
+		onCancel: [ : dialog |
+			"CmdCommandAborted signal"
+			cancelled := true.
+			dialog close ];
+		onAccept: [ :dialog | dialog close ];
+		openModal
```

Comment:
Ideally, this should be moved to the driver as well

Reply:
I'm not sure how to do it. Maybe for a next iteration? (I'm trying to not make this PR bigger)

Reply:
We should not have any logic in the command. This should be moved to the driver.

Reply:
Just move it there. 
The length of the PR is only important once we are feature complete :)